flink-optimizer/src/main/java/org/apache/flink/optimizer/operators/AbstractJoinDescriptor.java

```diff
@@ -142,9 +142,9 @@ public boolean areCompatible(RequestedGlobalProperties requested1, RequestedGlob
 			else if(produced1.getPartitioning() == PartitioningProperty.RANGE_PARTITIONED &&
 					produced2.getPartitioning() == PartitioningProperty.RANGE_PARTITIONED) {
 
-				// Return false anyway now, we need both the partition key and data distribution
-				// information to make sure whether the range partitions are equivalent.
-				return false;
+				return checkEquivalentFieldPositionsInKeyFields(
```

Comment:
Should this be done in the next issue you are working on? And i don't think "==" should be used here, as the data distribution from 2 input should not be the same object instance(they should be generated by deserialization most of time).